src/main.rs

```diff
@@ -130,16 +148,58 @@ fn main() -> Result<()> {
     });
 
     // Generate the temporary project
-    let mut command = process::Command::new("cargo");
-    command
-        .current_dir(&tmp_dir)
-        .args(&["init", "--name", project_name.as_str()]);
-    if cli.lib {
-        command.arg("--lib");
+    if let Some(maybe_branch) = cli.worktree_branch.as_ref() {
+        let mut command = process::Command::new("git");
+        command.args(&["worktree", "add"]);
+
+        match maybe_branch {
+            Some(branch) => command.arg(&tmp_dir.path()).arg(branch.as_str()),
+            None => command.arg("-d").arg(&tmp_dir.path()),
+        };
+        if !command.status().context("Could not start git")?.success() {
```

Comment:
This looks like a good use of `anyhow::ensure`

Reply:
Nice I didn't know that